src/EditorFeatures/TestUtilities/DispatcherExtensions.cs

```diff
@@ -9,26 +13,87 @@ public static class DispatcherExtensions
     {
         public static void DoEvents(this Dispatcher dispatcher)
         {
+            EnsureHooked();
             // A DispatcherFrame represents a loop that processes pending work
             // items.
             var frame = new DispatcherFrame();
             var callback = (Action<DispatcherFrame>)(f => f.Continue = false);
 
-            // Executes the specified delegate asynchronously.  When it is 
-            // complete mark the frame as complete so the dispatcher loop
-            // pops out (stops).
-            var operation = dispatcher.BeginInvoke(
-                DispatcherPriority.SystemIdle, callback, frame);
+            try
+            {
+                // Executes the specified delegate asynchronously.  When it is 
+                // complete mark the frame as complete so the dispatcher loop
+                // pops out (stops).
+                var operation = dispatcher.BeginInvoke(
+                    DispatcherPriority.SystemIdle, callback, frame);
+
+                // Start the loop.  It will process all items in the queue, then 
+                // will process the above callback.  That callback will tell the
+                // loop to then stop processing.
+                Dispatcher.PushFrame(frame);
+
+                if (operation.Status != DispatcherOperationStatus.Completed)
+                {
+                    operation.Abort();
+                }
+            }
+            catch (TargetParameterCountException ex)
+            {
+                var methodInfoBuilder = new StringBuilder();
+                methodInfoBuilder.AppendLine("Caught TargetParameterCountException in DoEvents. Printing MethodInfos in reverse execution order.");
+                PropertyInfo fullNameProperty = typeof(MethodBase).GetProperty("FullName", BindingFlags.NonPublic | BindingFlags.Instance);
```

Comment:
I dont think you actually meant all operation delegate but execution order of all nested operations? in that case, should you pop operation from delegateinfo once that operation is successfully executed?

Reply:
Yeah, you're right. I'll update it to pop when the delegate is successfully executed.